pylint/checkers/typecheck.py

```diff
@@ -1585,7 +1585,7 @@ def _check_invalid_sequence_index(self, subscript: nodes.Subscript):
             not isinstance(itemmethod, nodes.FunctionDef)
             or itemmethod.root().name != "builtins"
             or not itemmethod.parent
-            or itemmethod.parent.name not in SEQUENCE_TYPES
+            or itemmethod.parent.frame().name not in SEQUENCE_TYPES
```

Comment:
Not sure if we needed this. It's not tested because it only guards against a new conditional definition of a dunder method in a built-in package. I am fine with leaving it out.

Reply:
Do you know what is the cost of calling frame() ?

Reply:
Seems low. `frame` returns `frame` of `parent` or returns `self` if it is already a frame, so when there's no conditional statement before it's `FuncDef -> parent -> frame -> ClassDef` compared to `FuncDef -> parent -> ClassDef`. 

Experimentally profiling on Django:

```
grep _check_signature django_frame.profile
     1465    0.007    0.000    0.111    0.000 class_checker.py:1950(_check_signature)
grep _check_signature django_no_frame.profile
     1465    0.007    0.000    0.111    0.000 class_checker.py:1950(_check_signature)
```

```
grep \(frame\) django_frame.profile
      170    0.000    0.000    0.000    0.000 node_classes.py:4801(frame)
5846821/1771656    1.697    0.000    1.812    0.000 node_ng.py:311(frame)
     4554    0.000    0.000    0.000    0.000 scoped_nodes.py:1480(frame)
  1844282    0.118    0.000    0.118    0.000 scoped_nodes.py:2012(frame)
  2191882    0.115    0.000    0.115    0.000 scoped_nodes.py:3259(frame)
  5025701    0.270    0.000    0.270    0.000 scoped_nodes.py:865(frame)
grep \(frame\) django_no_frame.profile
      170    0.000    0.000    0.000    0.000 node_classes.py:4801(frame)
5846830/1771660    1.654    0.000    1.769    0.000 node_ng.py:311(frame)
     4554    0.000    0.000    0.000    0.000 scoped_nodes.py:1480(frame)
  1844287    0.118    0.000    0.118    0.000 scoped_nodes.py:2012(frame)
  2189318    0.115    0.000    0.115    0.000 scoped_nodes.py:3259(frame)
  5025600    0.272    0.000    0.272    0.000 scoped_nodes.py:865(frame)
```

Reply:
I think it's safe to keep even if we do not strictly needs it then.